### PR TITLE
Tslint prettier

### PR DESCRIPTION
**Pull request by TN1ck**
## Tslint
Finally Some linting is there! I also added a pre-commit hook to disallow any commits that contain linting errors.
I use the same config `/tslint.json` for frontend and backend.

## Prettier
I also added prettier, there are no hooks yet, but I formatted the code once with it.

I also fixed issues that were triggered by the new linting.

## Source Maps
As a bonus I added source-map support for the backend so one can see where an error actually occured.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "no-var-requires": false,
+    "arrow-parens": false,
+    "only-arrow-functions": [
+      false
+    ],
+    "member-access": false,
+    "member-ordering": [
+      "field-first"
+    ],
+    // TODO: these should be activated at one point
```
Comment (bennycode) [inline tslint.json]: There are no comments in JSON.

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "ordered-imports": [false],
```
Comment (bennycode) [inline tslint.json]: Does it need an array or would `"ordered-imports": false` be sufficient?

File: frontend/package.json [hunk 3]
```diff
@@ -12,12 +12,13 @@
...
     "http-proxy-middleware": "0.17.4",
     "node-sass": "4.5.2",
+    "prettier": "^1.4.4",
     "react-hot-loader": "3.0.0-beta.7",
     "sass-loader": "6.0.5",
     "style-loader": "0.17.0",
     "ts-loader": "2.0.3",
-    "tslint": "5.2.0",
-    "tslint-loader": "3.5.3",
+    "tslint": "^5.2.0",
```
Comment (bennycode) [inline frontend/package.json:20]: We should go for fix versions (`5.2.0`).

File: backend/package.json [hunk 4]
```diff
@@ -8,28 +8,30 @@
...
-    "tslint": "5.2.0",
-    "tslint-loader": "3.5.3",
     "typescript": "2.3.2",
     "webpack": "2.5.1"
   },
   "devDependencies": {
     "concurrently": "3.4.0",
     "nodemon": "1.11.0",
-    "sqlite3": "3.1.8"
+    "prettier": "^1.4.4",
```
Comment (bennycode) [inline backend/package.json]: As long as we don't use yarn I would always go for fixed versions.

File: backend/package.json [hunk 5]
```diff
@@ -45,7 +47,10 @@
     "build": "cross-env NODE_ENV=production && webpack",
     "start-prod": "cross-env NODE_PATH=./dist && node dist/main.js",
     "database:populate": "cross-env NODE_PATH=./dist && node dist/populate_database.ts.js",
-    "database:reset": "cross-env NODE_PATH=./dist && node dist/reset.ts.js"
+    "database:reset": "cross-env NODE_PATH=./dist && node dist/reset.ts.js",
+    "prettier:base": "prettier --parser typescript --single-quote --trailing-comma='all' --no-bracket-spacing",
```
Comment (bennycode) [inline backend/package.json]: Do we need 3 prettier scripts? Isn't just one (with `--write`) enough?)

**Review by bennycode — CHANGES_REQUESTED** [review]

File: tslint.json (hunk 1 above)
Comment (TN1ck) [reply tslint.json]: It works though as is normally done like this, see the tslint.json from the maintainers of tslint: https://github.com/palantir/tslint/blob/master/tslint.json

File: tslint.json (hunk 2 above)
Comment (TN1ck) [reply tslint.json]: Needs an array or at least the type-checking of the file says so

File: backend/package.json (hunk 5 above)
Comment (TN1ck) [reply backend/package.json]: I adopted this schema from somewhere else. It makes it a bit better looking imho

File: backend/package.json (hunk 4 above)
Comment (TN1ck) [reply backend/package.json]: Correct, there is npm-lock.json as well now :P

File: frontend/package.json (hunk 3 above)
Comment (TN1ck) [reply frontend/package.json:20]: Correct

**Review by bennycode — APPROVED** [review]

File: tslint.json (hunk 1 above)
Comment (bennycode) [reply tslint.json]: We should have more dignity than these TSLint barbarians. 😁 There are no comments allowed in JSON (https://jsonlint.com/ will show the truth). Let's be professional programmers who don't misuse well-defined standards. 🌵

File: tslint.json (hunk 2 above)
Comment (bennycode) [reply tslint.json]: ok!

File: frontend/package.json (hunk 3 above)
Comment (bennycode) [reply frontend/package.json:20]: Ok, then this needs pinning too. At the moment only backend properties are pinned.

File: backend/package.json (hunk 5 above)
Comment (bennycode) [reply backend/package.json]: But is there a need for `prettier:base` and `prettier:check`? Isn't just `prettier:write` enough?

**Review by bennycode — CHANGES_REQUESTED** [review]

**Review by bennycode — APPROVED** [review]

**Pull request merged**